lib/index.js

```diff
@@ -682,6 +683,13 @@ var Client = module.exports = function(config) {
                     basic = new Buffer(this.auth.username + ":" + this.auth.password, "ascii").toString("base64");
                     headers["Authorization"] = "Basic " + basic;
                     break;
+                case "netrc":
+                    var auth = netrc()["api.github.com"];
```

Comment:
The way it works in master right now as merged you need to add an entry in netrc for uploads.github.com in addition to api.github.com. I just added a test to make sure it works: https://github.com/mikedeboer/node-github/blob/master/examples/netrcAuth.js